gateway/crates/federated-server/src/config/hot_reload.rs

```diff
@@ -0,0 +1,124 @@
+use std::{collections::HashMap, fs, path::PathBuf, sync::OnceLock, time::Duration};
+
+use grafbase_telemetry::span::GRAFBASE_TARGET;
+use notify::{EventHandler, EventKind, PollWatcher, Watcher};
+use runtime::rate_limiting::GraphRateLimit;
+use tokio::sync::{mpsc, watch};
+
+use crate::Config;
+
+type RateLimitData = HashMap<String, GraphRateLimit>;
+
+pub(crate) enum RateLimitSender {
+    Watch(watch::Sender<RateLimitData>),
+    Mpsc(mpsc::Sender<RateLimitData>),
+}
+
+impl RateLimitSender {
+    fn send(&self, data: RateLimitData) -> crate::Result<()> {
```

Comment:
:zap: **Suggestion** 
Consider using `expect` instead of `unwrap` for error handling in the `send` method to provide more context in case of failure.